lib/addons/ack/index.ts

```diff
@@ -0,0 +1,108 @@
+// lib/certmanager_addon.ts
+import { ManagedPolicy } from 'aws-cdk-lib/aws-iam';
+import { Construct } from 'constructs';
+import merge from "ts-deepmerge";
+import { ClusterInfo, Values } from "../../spi";
+import { createNamespace, setPath } from "../../utils";
+import { HelmAddOn, HelmAddOnProps, HelmAddOnUserProps } from "../helm-addon";
+/**
+ * User provided option for the Helm Chart
+ */
+export interface AckAddOnProps extends HelmAddOnUserProps {
+    
+    /**
+     * Name of the ack controller
+     * @default iam-chart
+     */
+    name?: string;
+    /**
+     * Name of the ack controller Chart
+     * @default iam-chart
+     */
+    chart?: string;
+    /**
+     * Version of the ack controller
+     * @default v0.0.13
+     */
+    version?: string;
+    /**
+     * Release of ack controller
+     * @default iam-chart
+     */
+    release?: string;
+    /**
+     * Repository of ack controller
+     * @default oci://public.ecr.aws/aws-controllers-k8s/iam-chart
+     */
+    repository?: string;
+    /**
+     * Managed IAM Policy of the ack controller
+     * @default IAMFullAccess
+     */
+    managedPolicyName?: string;
+    /**
+     * To Create Namespace using CDK. This should be done only for the first time.
+     */    
+    createNamespace?: boolean;
+}
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+const defaultProps: HelmAddOnProps & AckAddOnProps = {

```

Comment:
Why `HelmAddOnProps &` and not just AckAddOnProps? I suggest keeping just AckAddOnProps, unless you see a good reason to do otherwise.

Reply:
This wont work and this line `super({...defaultProps, ...props});` fail.